experiments/post_layout_optimization/post_layout_optimization.cpp

```diff
@@ -0,0 +1,114 @@
+#include <lorina/lorina.hpp>                 // Verilog/BLIF/AIGER/... file parsing
+#include <mockturtle/io/verilog_reader.hpp>  // call-backs to read Verilog files into networks
+
+#include <cassert>
+#include <chrono>
+#include <cstdlib>
+#include <string>
+
+int main()  // NOLINT
+{
+    using gate_lyt =
+        fiction::gate_level_layout<fiction::clocked_layout<fiction::tile_based_layout<fiction::cartesian_layout<>>>>;
+
+    experiments::experiment<std::string, uint32_t, uint32_t, uint32_t, uint64_t, uint64_t, uint64_t, uint64_t, uint64_t,
+                            uint64_t, uint32_t, uint32_t, uint64_t, uint64_t, double, double, double, std::string>
+        optimization_exp{"optimization",
+                         "benchmark",
+                         "inputs",
+                         "outputs",
+                         "initial nodes",
+                         "ortho layout width (in tiles)",
+                         "ortho layout height (in tiles)",
+                         "ortho layout area (in tiles)",
+                         "optimized layout width (in tiles)",
+                         "optimized layout height (in tiles)",
+                         "optimized layout area (in tiles)",
+                         "gates",
+                         "wires",
+                         "critical path",
+                         "throughput",
+                         "runtime ortho (in sec)",
+                         "runtime optimization (in sec)",
+                         "improvement (%)",
+                         "equivalent"};
+
+    // stats for SMT-based physical design
+    fiction::orthogonal_physical_design_stats orthogonal_stats{};
+
+    static constexpr const uint64_t bench_select =
+        fiction_experiments::all & ~fiction_experiments::epfl & ~fiction_experiments::iscas85;
+
+    for (const auto& benchmark : fiction_experiments::all_benchmarks(bench_select))
+    {
+        fmt::print("[i] processing {}\n", benchmark);
+
+        fiction::technology_network network{};
+
+        const auto read_verilog_result =
+            lorina::read_verilog(fiction_experiments::benchmark_path(benchmark), mockturtle::verilog_reader(network));
+        assert(read_verilog_result == lorina::return_code::success);
+
+        // perform layout generation with an OGD-based heuristic algorithm
+        auto gate_level_layout = fiction::orthogonal<gate_lyt>(network, {}, &orthogonal_stats);
+
+        //  compute critical path and throughput
+        fiction::critical_path_length_and_throughput_stats cp_tp_stats{};
+        fiction::critical_path_length_and_throughput(gate_level_layout, &cp_tp_stats);
+
+        // calculate bounding box
+        const auto bounding_box_before = fiction::bounding_box_2d(gate_level_layout);
+
+        const auto width  = bounding_box_before.get_x_size();
+        const auto height = bounding_box_before.get_y_size();
+
+        const auto begin = std::chrono::steady_clock::now();
```

Comment:
Runtime calculations should be done within the algorithm itself in a statistics object. See `orthogonal.hpp` for an example of how to implement this.